### PR TITLE
fix(elastic): inconsistent offset on bulk index error and improve bulk performance

### PR DESCRIPTION
**Pull request by alphinside**
Goals:

On bulk index submit, if there is error occured ( for example if data type provided is wrong ( in test, I give example by providing int64 data inserted into int32 column )), it will raise BulkIndexError, however the documents is still inserted to the main index but due to error raises, the offset is not updated.

In this PR, I try to fix the issue and also increase the performance by using parallel_bulk and set raise_on_error to False, then return the index error info to the caller

speed benchmark result :

1000 Docs, emb dim = 3, average of 20 runs

with parallel bulk, 4 thread = 15.89 s
with bulk = 21.34 s

- ...
- ...
- [x] check and update documentation, if required. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines)

### CODE REVIEW

File: docarray/array/storage/elastic/backend.py [hunk 1]
```diff
@@ -159,7 +159,12 @@ def _build_client(self):
         return client
 
     def _send_requests(self, request):
-        bulk(self._client, request)
+        failed_index = []
+        for success, info in parallel_bulk(self._client, request, raise_on_error=False):
+            if success is not True:
```
Comment (JoanFM) [inline docarray/array/storage/elastic/backend.py:164]:
```suggestion
            if not success:
```

**Review by JoanFM — CHANGES_REQUESTED** [review]
Are we sure this `not raise on error behavior` is compatible with the rest of the backends?

File: docarray/array/storage/elastic/seqlike.py [hunk 2]
```diff
@@ -59,19 +59,27 @@ def __repr__(self):
...
+            failed_index.extend(self._send_requests(batch))
             self._refresh(self._config.index_name)
 
+        return failed_index
+
     def extend(self, docs: Iterable['Document']):
         docs = list(docs)
-        self._upload_batch(docs)
+        failed_index = self._upload_batch(docs)
+        failed_ids = [index['_id'] for index in failed_index]
```
[the remark this answers is not in the document]
Comment (alphinside) [reply docarray/array/storage/elastic/seqlike.py:78]: yes, thats also what I thought, any reference of logger that you guys use, or just use the standard `logging` one?

[the remark this answers is not in the document]
Comment (hanxiao) [reply docarray/array/storage/elastic/seqlike.py:78]: This is a very customized behavior. I wouldn't recommend. To me this is basically play a ding sound on every success insert, which does not make it feel like Python List container.

Notice that the design principle of DocArray package and DocumentArray is making it a List-Like object.

This customized behavior violates this design principle.

**Review by hanxiao — CHANGES_REQUESTED** [review]
https://github.com/jina-ai/docarray/pull/422#discussion_r909507152

File: docarray/array/storage/elastic/seqlike.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alphinside) [reply docarray/array/storage/elastic/seqlike.py:78]: will be waiting @JoanFM decision on this, whether we treat extend operation as atomic or not

**Review by JoanFM — CHANGES_REQUESTED** [review]
I have been reading the code, and what seems more coherent with rest of backends would be to `raise` an Exception (without reverting the proper appends) and informing via warning what `Docs` were not able to be appended.

It seems to me that is the more `SeqLike` behavior there is. What do you think @alphinside ?

**Comment by alphinside** [comment]
> I have been reading the code, and what seems more coherent with rest of backends would be to `raise` an Exception (without reverting the proper appends) and informing via warning what `Docs` were not able to be appended.
> 
> It seems to me that is the more `SeqLike` behavior there is. What do you think @alphinside ?

yes , I agree will update on that aspect

**Review by JoanFM — APPROVED** [review]
LGTM!

**Pull request merged**